### PR TITLE
Enable migration from accounts to keys

### LINKED ISSUE #128: Test key migration code
The Keyguard needs to correctly migrate between legacy key store and new key store, when instructed by the Accounts Manager.

On iOS, this means detecting the migration cookie and migrating on next top-level request.

### PR DESCRIPTION
**Pull request by sisou**
Also includes
- Additional iframe `hasKeys` boolean method
- refactoring `BrowserDetection.isIos() => BrowserDetection.isIOS()`
- Test fixes
- More refactorings

Resolves #128.

### CODE REVIEW

File: src/request/iframe/IFrameApi.js [hunk 1]
```diff
@@ -8,25 +8,41 @@
 class IFrameApi {
     /**
      * @param {Rpc.State | null} state
-     * @param {boolean} [listFromLegacyStore] - Deprecated, only for database migration
```
Comment (nibhar) [inline src/request/iframe/IFrameApi.js]: I am not sure, but I'd personally rather have an additional function
```
/**javascript
 * @param {Rpc.State | null} state
 * @deprecated
 * @returns {Promise<KeyInfoObject[]>}
*/
async listFromLegacyStore(state) {
```
than the same function with different parameters being deprecated or not.
Also i think @deprecated should be added to deprecated methods, which also is not exactly possible with the deprecated and at the same time not deprecated methods.

File: src/request/iframe/IFrameApi.js [hunk 2]
```diff
@@ -8,25 +8,41 @@
...
 
-        const keyInfos = await KeyStore.instance.list();
+        const keyInfos = await this._getKeys();
         return keyInfos.map(ki => ki.toObject());
     }
 
     /**
      * @param {Rpc.State | null} state
-     * @returns {Promise<void>}
+     * @param {boolean} [inLegacyStore] - Deprecated, only for database migration
```
Comment (nibhar) [inline src/request/iframe/IFrameApi.js]: same as above

File: src/common.js [hunk 3]
```diff
@@ -1,29 +1,25 @@
...
 async function loadNimiq() {
-    try {
+    // eslint-disable-next-line no-return-assign
+    return __nimiqLoaded || (__nimiqLoaded = new Promise(async resolve => {
         // Load web assembly encryption library into browser (if supported)
         await Nimiq.WasmHelper.doImportBrowser();
+
         // Configure to use test net for now
         Nimiq.GenesisConfig.test();
-    } catch (e) {
```
Comment (nibhar) [inline src/common.js]: In what case does `Nimiq.WasmHelper.doImportBrowser()` throw? Can we handle it, do we need to handle it?

Comment (sisou) [reply src/common.js]: `doImportBrowser()` throws when it's already been called before. Which should not happen with this singleton promise:

From https://github.com/nimiq-network/core/blob/master/src/main/generic/utils/WasmHelper.js#L5
```javascript
        if (WasmHelper._importStarted) {
            Log.e(WasmHelper, 'doImportBrowser invoked twice');
            return;
        }
```

File: client/types/KeyguardRequest.d.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ declare namespace KeyguardRequest {
         type: Key.Type;
         encrypted: boolean;
         hasPin: boolean;
-        userFriendlyId?: string;
+        legacyAccount?: { label: string, address: Uint8Array };
     }
```
[the remark this answers is not in the document]
Comment (sisou) [reply client/types/KeyguardRequest.d.ts]: No, only when returned from the `listLegacyAccounts` method. But you are right, maybe it should have another type then.

File: tests/lib/CookieJar.spec.js [hunk 5]
```diff
@@ -1,5 +1,6 @@
 /* global Dummy */
 /* global CookieJar */
+
```
[the remark this answers is not in the document]
Comment (sisou) [reply tests/lib/CookieJar.spec.js]: I do what I can :stuck_out_tongue:

File: types/Keyguard.d.ts [hunk 6]
```diff
@@ -41,7 +41,7 @@ type KeyInfoObject = {
     type: Key.Type
     encrypted: boolean
     hasPin: boolean
-    userFriendlyId?: string
+    legacyAccount?: { label: string, address: Uint8Array };
 }
```
[the remark this answers is not in the document]
Comment (sisou) [reply types/Keyguard.d.ts]: Yes, you are right. We should.

File: src/request/create/Create.js [hunk 7]
```diff
@@ -162,6 +162,12 @@ class Create {
         // go to start page
         window.location.hash = Create.Pages.CHOOSE_IDENTICON;
         this._identiconSelector.generateIdenticons();
+
+        // XXX: Is the following necessary, or is the CryptoWorker already
+        // instanced when generating addresses for identicons?
```
[the remark this answers is not in the document]
Comment (sisou) [reply src/request/create/Create.js]: I haven't checked specifically, but this was a comment stored in my stash for a long time now, so I just put it in here for now. It's a performance optimization, to pre-load the CryptoWorker to reduce encryption time after first password entry.

**Pull request merged**